src/tests/aes.rs

```diff
@@ -500,14 +514,10 @@ fn test_aes_operations() {
                 Err(e) => panic!("{}", e),
             };
 
-        let ciphertext = match get_test_data(session, testname, "ciphertext") {
-            Ok(vec) => vec,
-            Err(ret) => return assert_eq!(ret, CKR_OK),
-        };
-        let plaintext = match get_test_data(session, testname, "plaintext") {
-            Ok(vec) => vec,
-            Err(ret) => return assert_eq!(ret, CKR_OK),
-        };
+        let ciphertext = hex::decode("4154c0be71072945d8156f5f046d198d")
+            .expect("Failed to decode ciphertext");
+        let plaintext = hex::decode("8b2b1b22f733ac09d1196d6be6a87a72")
+            .expect("Failed to decode plaintext");
```

Comment:
What about turning get_test_data to be a function like you did for get_gcm_test_data() ?
It can be a function that return data from an array of test data base on an index or something.
Also in the commit message CKA_DATA -> CKO_DATA

Reply:
I did not like either approach. With the `get_test_data`, the hex blocks get very indented, but huge blobs inside of function make the reading harder. In case of AES, the values are quite short so I considered this ok.